ADAL/src/ADAuthenticationParameters+Internal.m

```diff
@@ -168,6 +168,150 @@ + (NSDictionary *)extractChallengeParameters:(NSString *)headerContents
         *error = adError;
     }
     return nil;
+
+}
+
++ (NSDictionary *)extractChallengeParameters:(NSString *)headerContents
+                                       error:(ADAuthenticationError * __autoreleasing *)error;
+{
+    __block ADAuthenticationError* adError = nil;
+    NSDictionary *parameters = nil;
+    
+    if ([NSString adIsStringNilOrBlank:headerContents])
+    {
+        adError = [self invalidHeader:headerContents];
+    }
+    else
+    {
+        NSString *bearerChallendge = [self extractBearerChallenge:headerContents];
+        if (!bearerChallendge) {
+            adError = [self invalidHeader:headerContents];
+        } else {
+            parameters = [self parseBearerChallenge:bearerChallendge error:&adError];
+        }
+    }
+    
+    if (error)
+    {
+        *error = adError;
+    }
+    
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the useless trailing whitespaces at the end of this line. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=objc%3AS1131)